packages/core/src/combinator/timeslice.js

```diff
@@ -3,111 +3,102 @@
 /** @author John Hann */
 
 import Pipe from '../sink/Pipe'
-import { disposeAll } from '@most/disposable'
+import { disposeBoth } from '@most/disposable'
 import { join } from './chain'
+import { merge } from './merge'
+import { never } from '../source/never'
+
+const timeBounds = (min, minSignal, maxSignal) =>
+  ({ min, minSignal, maxSignal })
+
+const minBounds = (minSignal) =>
+  timeBounds(Infinity, minSignal, never())
+
+const maxBounds = (maxSignal) =>
+  timeBounds(0, never(), maxSignal)
+
+const mergeTimeBounds = (b1, b2) =>
+  ({
+    min: Math.max(b1.min, b2.min),
+    minSignal: merge(b1.minSignal, b2.minSignal),
+    maxSignal: merge(b1.maxSignal, b2.maxSignal)
```

Comment:
For these `merge`s to be maximally efficient, we could make `merge` discard `never` (as it does with `empty`).  I don't know that it matters all that much, since `never` is totally a noop stream (even moreso than `empty`!), and returns `disposeNone` which the disposable fusion (also in this PR) will handle efficiently.

Reply:
Hmmm, no, discarding `never` would not be correct for `merge`.  `never` would affect the end time of the resulting merged stream.  For example, while these contain the same event (value `1` at time zero), the first ends and the second doesn't: `now(1)` and `merge(now(1), never())`.

So, I think I'm content just to let the "noop-ness" of never, plus disposable fusion take care of the situation.

FWIW, Merge could squash multiple `never`s into a single `never`, but that doesn't seem worth it for the same reasons.  E.g. with disposable fusion, `mergeArray([never(), never(), never(), ...])` ends up just creating a bunch of `disposeNone`s, which disposable fusion would then squash into a single `disposeNone`.